### DISCUSSION TITLE
Help needed connecting Celestron 114GT (Pre 2012) HS to Stellarium v1.1

### DISCUSSION THREAD

**Discussion by pillai008** [Stellarium/stellarium#2924] [Q&A]
I just cannot coax any movement from the mount with Stellarium after a CNTRL+O, slew or sync. 
The HC is a non-LCD type shipped by Celestron to overcome initial issues of serial port bugs with the LCD version.
Using the 9 Pin Serial-RJ11 serial cable that came with the telescope into the HC and a USB to 9 Pin Serial cable to connect to the Debian linux v6.0.0.6 PC USB port. The port appears as /dev/ttyUSB0 and using JNow coordinate system in Stellarium.
Starting Stellarium with sudo and --on-sandbox to enable access to the serial port.

Stellarium log says:

[TelescopeControl] port name: "ttyUSB0" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "ttyUSB0" ; vendor identifier: 1659 ; product identifier: 9155
socket: Connection refused
[TelescopeControl] port name: "ttyUSB0" ; vendor identifier: 1659 ; product identifier: 9155
connectionType: TelescopeControl::ConnectionInternal  initString: "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:/dev/ttyUSB0:500000"
Creating telescope "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:/dev/ttyUSB0:500000" ; name/type/equinox/params: "Celestron Nexstar 114GT" "TelescopeServerNexStar" JNow "/dev/ttyUSB0:500000"
TelescopeClientDirectNexStar parameters: port, time_delay: "/dev/ttyUSB0" 500000
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
[TelescopeControl] port name: "ttyUSB0" ; vendor identifier: 1659 ; product identifier: 9155

![Screenshot from 2022-12-20 09-27-20](https://user-images.githubusercontent.com/44720086/208750512-6e1e5cd5-24db-4d6c-9d75-bf82489c0361.png)

HC is non-upgradebale:
![20221221_063812](https://user-images.githubusercontent.com/44720086/208752683-959a476d-f274-447a-9791-2180b8a12097.jpg)

**Comment by gzotti** [comment]
If this is your computer, try getting access to the port by configuring your user settings properly, not by running Stellarium with sudo. 
Do I also read Debian 6? Seriously? This is obsolete since 2016. Did Stellarium 0.16 work properly?

**Reply by pillai008** [reply]
Thanks for the prompt feedback!
Debian is at the latest version
![Screenshot from 2022-12-21 07-06-20](https://user-images.githubusercontent.com/44720086/208757769-2a2d86ef-2f1f-4db8-bdb1-31c7166c48c3.png)

I got Stellarium to run as myself and not root by adding myself to the dialout group, so now I have access to the serial port - (sudo usermod -a -G dialout {mylogin})
So now I start this as ./Stellarium-1.1.1-x86_64.AppImage 
Same results after doing all this. 

I started with Stellarium 0.22 with the same results, which prompted me to upgrade to 1.1
BTW this is the first time I am trying this, so there is no history of having worked before. I can try 0.16, is this suggestion based on your assumption of old Debian version? I ran into compilation issues with 0.16, so looks to be a nogo there to test.

**Reply by gzotti** [reply]
Ah no, 0.16 was based on that false assumption. Version 1.1 should work. We have indications that Qt5-based builds may work better, though. We have occasional reports of problems with the telescope plugin, but we need a few advanced users/programmers who would be able to actually identify and fix them.

**Reply by pillai008** [reply]
willing to troubleshoot this further if you can guide where to look. Is there any switches or tweaks to unravel if any comms is happening at all.

**Comment by pillai008** [comment]
On Windows 10 platform, I tried the following:
The serial port appears as below:
![Capture1](https://user-images.githubusercontent.com/44720086/208787861-fa3504a8-36fb-4857-96f2-25e2bd2d4fca.PNG)

Starting Stellarium has a different experience, it just cannot connect to it.
![Capture2](https://user-images.githubusercontent.com/44720086/208788149-0aa27928-c13f-4239-90a6-59ae32c17ddc.PNG)

Logs in  AppData\Roaming\Stellarium show this:
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
connectionType: TelescopeControl::ConnectionInternal  initString: "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000"
Creating telescope "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000" ; name/type/equinox/params: "Celestron Nexstar 114GT" "TelescopeServerNexStar" JNow "COM4:500000"
TelescopeClientDirectNexStar parameters: port, time_delay: "COM4" 500000
ERROR creating TelescopeClientDirectNexStar: cannot open serial device "COM4"
TelescopeClient::create(): Unable to create a telescope client.
Unloaded plugin "TelescopeControl"

If I start Celestron's NexRemote program on Windows 10, I can connect to the telescope successfully.
with these setting:
![Capture3](https://user-images.githubusercontent.com/44720086/208790323-aebff310-6bc4-427e-aada-2e8a6d7e003f.PNG)
![Capture3](https://user-images.githubusercontent.com/44720086/208789335-be1d45d9-1e9b-48b8-b890-3bb935848eda.PNG)

This confirms the hardware connections and the wires are connected up fine. I can slew and toggle through the menu options using the soft remote on the PC.

The culprit seems to be Stellarium comms to telescope, even stranger why the Windows version cannot open the COM port too???
Any clues?

**Reply by gzotti** [reply]
Have you tried with the Qt5-based version 1.1 (a.k.a. 0.22.4)? Else I am running out of ideas, which is why I hope for "someone out there" taking over on this.

**Reply by pillai008** [reply]
i am using that version with no luck
![Screenshot from 2022-12-21 13-09-26](https://user-images.githubusercontent.com/44720086/208804587-04ed16c4-8063-462d-825f-e918cf74878e.png)

**Reply by pillai008** [reply]
My assumption was incorrect on Windows platform, all qt versions both 5 and 6 have the same initial splash.  
So I downloaded the correct build with qt5 and installed it now.
![CaptureB](https://user-images.githubusercontent.com/44720086/208816608-0f61503d-96e2-43cf-a890-4cb10ff8d2bd.PNG)
This version now connects to the telescope!
![CaptureA](https://user-images.githubusercontent.com/44720086/208816735-eb883e42-9b3d-4347-9e4c-59a93eaedb21.PNG)

So we now have the same behavior on both Windows and Linux platforms, now the plugin connects, but that all. No control.
Logs show
Loaded plugin "TelescopeControl"
[TelescopeControl] Only embedded telescope servers are available.
[TelescopeControl] Adding device model: "Meade AutoStar compatible" "Any telescope or telescope mount compatible with Meade's AutoStar controller." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Meade LX200 (compatible)" "Any telescope or telescope mount compatible with Meade LX200." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Meade ETX70 (#494 Autostar, #506 CCS)" "Meade's ETX70 with the #494 Autostar controller and the #506 Connector Cable Set." "TelescopeServerLx200" 1500000
[TelescopeControl] Adding device model: "Losmandy G-11" "Losmandy's G-11 telescope mount." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Wildcard Innovations Argo Navis (Meade mode)" "Wildcard Innovations' Argo Navis DTC in Meade LX200 emulation mode." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Celestron NexStar (compatible)" "Any telescope or telescope mount compatible with Celestron NexStar." "TelescopeServerNexStar" 500000
[TelescopeControl] Adding device model: "Sky-Watcher SynScan (version 3 or later)" "Any Sky-Watcher mount that uses version 3 or later of the SynScan hand controller." "TelescopeServerNexStar" 500000
[TelescopeControl] Adding device model: "Sky-Watcher SynScan AZ GOTO" "The Sky-Watcher SynScan AZ GOTO mount used in a number of telescope models." "TelescopeServerNexStar" 500000
connectionType: TelescopeControl::ConnectionInternal  initString: "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000"
Creating telescope "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000" ; name/type/equinox/params: "Celestron Nexstar 114GT" "TelescopeServerNexStar" JNow "COM4:500000"
TelescopeClientDirectNexStar parameters: port, time_delay: "COM4" 500000
[TelescopeControl] Loaded successfully 1 telescopes.
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "COM5" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM6" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCA2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCB2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "COM5" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM6" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCA2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCB2" ; vendor identifier: 0 ; product identifier: 0
Initializing planets GL shaders... 
[2] "Australasia" 1
Loaded 39 / 39 references
SlewDialog::loadPointsFromFile(): No points loaded. File is missing: "C:\\Users\\sriku\\AppData\\Roaming\\Stellarium\\modules\\TelescopeControl\\points.json"
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "COM5" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM6" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCA2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCB2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM4" ; vendor identifier: 1659 ; product identifier: 9155
[TelescopeControl] port name: "COM5" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "COM6" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCA2" ; vendor identifier: 0 ; product identifier: 0
[TelescopeControl] port name: "CNCB2" ; vendor identifier: 0 ; product identifier: 0
connectionType: TelescopeControl::ConnectionInternal  initString: "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000"
Creating telescope "Celestron Nexstar 114GT:TelescopeServerNexStar:JNow:COM4:500000" ; name/type/equinox/params: "Celestron Nexstar 114GT" "TelescopeServerNexStar" JNow "COM4:500000"
TelescopeClientDirectNexStar parameters: port, time_delay: "COM4" 500000
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Telescope "Celestron Nexstar 114GT" does not support AbortSlew()!
Unloaded plugin "TelescopeControl"

I installed the Trial version of Serial Port Monitor from https://www.eltima.com/ and sure enough I can see Stellarium open the port, but no data chatter unlike the session I had with the Celestron NexRemote program running.
Stellarium Port Monitor output:
![CaptureC](https://user-images.githubusercontent.com/44720086/208818011-653ea5bf-7841-4e54-bc7e-dbd4d56584d8.PNG)

Celestron NexRemote Port Monitor output:
![CaptureD](https://user-images.githubusercontent.com/44720086/208818034-e232ffe1-bb21-41a8-929e-0df472e6e1c1.PNG)

**Comment by gzotti** [comment]
Continued in #2934 which has the same problem about not being able to open a COM port.
